### ISSUE TITLE
Easy pairing with Loop

### ISSUE THREAD

**Issue by ps2** [nightscout/cgm-remote-monitor#7068]
Adding Nightscout to Loop currently involves the user typing in a long URL and an api key on their phone, which is an error prone process. QR-Code style pairing would make setting up Nightscout a breeze. Nightscout could have a feature to present a qr-code containing the site url and an access key (ideally separate from the main key) that would give Loop access to upload data. I'd be happy to work on the Loop side of this pairing process, if someone were to implement the NS side.

**Comment by bjornoleh** [comment]
@bewest , I believe you have mentioned an already existing QR code setup for Nightscout elsewhere. Perhaps you could describe tgat feature here?

**Comment by bewest** [comment]
Fantastic, apologies for my delay while traveling.
XDrip and the origional Android Uploader supported this feature.  Since then, it is also used by 640 Medtronic Uploader, XDrip, as well as DiaBox and potentially others.  T1Pal provides all customers with these QR codes automaticaly.
The QR code embeds a json object payload, containing a list of Nightscout REST endpoints.  Each Nightscout endpoint is URL encoded, with the API Secret encoded in the "user auth" portion of the URL.  Notably, xdrip requires the path segment `/api/v1` to be included in the URL.

URL format: `scheme://API_SECRET@fqdn`.  Since Android Uploader, it's been extremely beneficial to retain the ability to upload to multiple Nightscouts; for this reason the json object allows a list, although in most instances, it is a list of one endpoint.

The autoconfigure page will produce valid QR codes;
https://nightscout.github.io/pages/configure/
For example, given (fake) API Secret: `abcdefghijklnop` and Nightscout URL `https://afullyqualifiedname.not`, the widget will produce a QR code embedding the following document:
```
{"rest":{"endpoint":["https://abcdefghijklmnop@afullyqualifiedname.not"]}}
```
Hope this helps!

**Comment by ps2** [comment]
Yeah, this part of the idea, but it's still making the user do a lot of manual steps. NS should be able to show the QR code itself, without the user having to copy/paste the url and secret (!) into a third party site.
